iDiff/util/tar_utils_test.go

```diff
@@ -0,0 +1,102 @@
+package tarUtil
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func TestUnTar(t *testing.T) {
+	testCases := []struct {
+		descrip  string
+		tarPath  string
+		target   string
+		expected string
+		err      bool
+	}{
+		{
+			descrip:  "Tar with files",
+			tarPath:  "testTars/la-croix1.tar",
+			target:   "testTars/la-croix1",
+			expected: "testTars/la-croix1-actual",
+		},
+		{
+			descrip:  "Tar with folders with files",
+			tarPath:  "testTars/la-croix2.tar",
+			target:   "testTars/la-croix2",
+			expected: "testTars/la-croix2-actual",
+		},
+		{
+			descrip:  "Tar with folders with files and a tar file",
+			tarPath:  "testTars/la-croix3.tar",
+			target:   "testTars/la-croix3",
+			expected: "testTars/la-croix3-actual",
+		},
+	}
+	for _, test := range testCases {
+		err := UnTar(test.tarPath, test.target)
+		if err != nil && !test.err {
+			t.Errorf("Got unexpected error: %s", err)
+		}
+		if err == nil && test.err {
+			t.Errorf("Expected error but got none: %s", err)
+		}
+		if !dirEquals(test.expected, test.target) {
+			t.Errorf("Directory created not correct structure.")
+		}
+		os.RemoveAll(test.target)
+	}
+
+}
+
+func TestIsTar(t *testing.T) {
+	testCases := []struct {
+		input    string
+		expected bool
+	}{
+		{input: "/testTar/la-croix1.tar", expected: true},
+		{input: "/testTar/la-croix1-actual", expected: false},
+	}
+	for _, test := range testCases {
+		actual := isTar(test.input)
+		if test.expected != actual {
+			t.Errorf("Expected: %t but got: %t", test.expected, actual)
+		}
+	}
+}
+
+func TestExtractTar(t *testing.T) {
+	tarPath := "testTars/la-croix3.tar"
+	target := "testTars/la-croix3"
+	expected := "testTars/la-croix3-full"
+	err := ExtractTar(tarPath)
+	if err != nil {
+		t.Errorf("Got unexpected error: %s", err)
+	}
+	if !dirEquals(expected, target) || !dirEquals(target, expected) {
+		t.Errorf("Directory created not correct structure.")
+	}
+	os.RemoveAll(target)
+
+}
+
+func dirEquals(actual string, path string) bool {
+	files1, _ := ioutil.ReadDir(actual)
+
+	for _, file := range files1 {
+		if _, ok := os.Stat(path + file.Name()); ok == nil {
```

Comment:
can you use [filepath.Join()](https://golang.org/pkg/path/filepath/#Join) here? just to be safe

also in the few places below